modules/bamutil/trimbam/main.nf

```diff
@@ -0,0 +1,44 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BAMUTIL_TRIMBAM {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::bamutil=1.0.15" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bamutil:1.0.15--h2e03b76_1"
+    } else {
+        container "quay.io/biocontainers/bamutil:1.0.15--h2e03b76_1"
+    }
+
+    input:
+    tuple val(meta), path(inbam), val(trim_left), val(trim_right)
```

Comment:
```suggestion
    tuple val(meta), path(bam), val(trim_left), val(trim_right)
```

Reply:
Are `trim_left` and `trim_right` sample-specific or would they be set at a pipeline-level? If the latter then you would do something like this instead:

```
input:
tuple val(meta), path(bam)
val trim_left
val trim_right
```